File: qc-core/src/main/java/com/eager/questioncloud/core/domain/payment/QuestionPaymentHistory.java
package com.eager.questioncloud.core.domain.payment;

import com.eager.questioncloud.core.domain.coupon.CouponType;
import com.eager.questioncloud.core.domain.question.Subject;
import java.time.LocalDateTime;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class QuestionPaymentHistory {
    private String paymentId;
    private List<OrderQuestion> orders;
    private QuestionPaymentCoupon coupon;
    private int amount;
    private Boolean isUsedCoupon;
    private QuestionPaymentStatus status;
    private LocalDateTime createdAt;

    @Getter
    @AllArgsConstructor
    public static class QuestionPaymentCoupon {
        private String title;
        private CouponType couponType;
        private int value;
    }

    @Getter
    @AllArgsConstructor
    public static class OrderQuestion {
        private Long questionId;
        private int amount;
        private String title;
        private String thumbnail;
        private String creatorName;
        private Subject subject;
        private String mainCategory;
        private String subCategory;
    }
}

File: qc-payment/src/main/kotlin/com/eager/questioncloud/payment/domain/QuestionPaymentHistory.kt
package com.eager.questioncloud.payment.domain

import com.eager.questioncloud.payment.enums.QuestionPaymentStatus
import com.eager.questioncloud.question.dto.QuestionInformation
import java.time.LocalDateTime

class QuestionPaymentHistory(
    val orderId: String,
    val userId: Long,
    val orders: List<QuestionPaymentHistoryOrder>,
    val coupon: QuestionPaymentCoupon?,
    val amount: Int,
    val isUsedCoupon: Boolean,
    val status: QuestionPaymentStatus,
    val createdAt: LocalDateTime,
) {

    companion object {
        @JvmStatic
        fun create(
            questionPayment: QuestionPayment,
            questions: List<QuestionInformation>
        ): QuestionPaymentHistory {
            val orders = questions.stream()
                .map { question: QuestionInformation ->
                    QuestionPaymentHistoryOrder.from(question)
                }
                .toList()

            return QuestionPaymentHistory(
                questionPayment.order.orderId,
                questionPayment.userId,
                orders,
                questionPayment.questionPaymentCoupon,
                questionPayment.amount,
                questionPayment.isUsedCoupon(),
                questionPayment.status,
                questionPayment.createdAt,
            )
        }
    }
}
